app/utils/wellknown.js

```diff
@@ -0,0 +1,110 @@
+// thanks to npm well-known-wallets-hns 
+import https from "https";
+import hdns from "hdns";
+
+const init = hsd => hdns.setServers(['103.196.38.38', '103.196.38.39', '103.196.38.40'] );
+
+class TLSAError extends Error {
+    constructor(message = 'Invalid TLSA', code = 'EINSECURE') {
+        super(message);
+        this.code = code;
+    }
+}
+
+const verifyTLSA = async (socket, host) => {
+    const cert = socket.getPeerCertificate(false);
+    const tlsa = await hdns.resolveTLSA(host, 'tcp', 443);
+
+    try {
+        const valid = hdns.verifyTLSA(tlsa[0], cert.raw);
```

Comment:
DANE is valid if any TLSA is valid, not just the first.

Reply:
OK. I'll modify these.